### ISSUE TITLE
Cannot read property 'bindingStartIndex' of null when using Library

### ISSUE THREAD

**Issue by davidbilge** [angular/angular#35586]
# 🐞 bug report

### Affected Package
Not sure. Maybe this is somewhere in @angular/cli or somewhere related to Webpack.

### Is this a regression?
Yes, the previous version in which this bug was not present was Angular 8

### Description
When creating a library as described in [the documentation](https://angular.io/guide/creating-libraries) and using it in an empty app, I get the following error at runtime:

    ERROR TypeError: Cannot read property 'bindingStartIndex' of null
    at Module.ɵɵelementStart (core.js:20997)
    at MyLibComponent_Template (my-lib.js:19)
    at executeTemplate (core.js:11930)
    at renderView (core.js:11716)
    at renderComponent (core.js:13232)
    at renderChildComponents (core.js:11519)
    at renderView (core.js:11742)
    at renderComponent (core.js:13232)
    at renderChildComponents (core.js:11519)
    at renderView (core.js:11742)

I do not see any errors while compiling.

## 🔬 Minimal Reproduction
I did the following:

- Create a `my-libs` project: `ng new my-libs --create-application=false`
- Create a library: `cd my-libs` and `ng generate library my-lib`
- Create an app next to this: `cd ..` and `ng new my-app`
- Add the library as a dependency of the app: Add `"my-lib": "file:../my-libs/dist/my-lib"` to the `dependencies` in the app's `package.json` and run `npm install`
- Import the `MyLibModule` in the `AppModule` (imported via `import { MyLibModule } from 'my-lib';`)
- Use the component in the `AppComponent`: `<lib-my-lib></lib-my-lib>`
- Build the library: `cd my-libs` and `ng build --prod`
- Run the app: `cd ..` and `ng serve`
- Point browser at `http://localhost:4200/`
- Open Developer Tools to look at the console

The resulting project is contained in this zip-file [repro.zip](https://github.com/angular/angular/files/4231679/repro.zip). I removed the `node_modules`, so please do

    $ cd my-libs
    $ npm install
    $ ng build --prod
    $ cd ..
    $ cd my-app
    $ npm install
    $ ng serve

## 🔥 Exception or Error
<pre><code>
    ERROR TypeError: Cannot read property 'bindingStartIndex' of null
    at Module.ɵɵelementStart (core.js:20997)
    at MyLibComponent_Template (my-lib.js:19)
    at executeTemplate (core.js:11930)
    at renderView (core.js:11716)
    at renderComponent (core.js:13232)
    at renderChildComponents (core.js:11519)
    at renderView (core.js:11742)
    at renderComponent (core.js:13232)
    at renderChildComponents (core.js:11519)
    at renderView (core.js:11742)
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 9.0.2
Node: 12.13.0
OS: darwin x64

Angular: 
... 
Ivy Workspace: 

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.900.2
@angular-devkit/core         9.0.2
@angular-devkit/schematics   9.0.2
@schematics/angular          9.0.2
@schematics/update           0.900.2
rxjs                         6.5.3
</code></pre>

**Comment by Shortyshort80** [comment]
I have the same issue. I have different workspaces for library and app.

**Comment by ioulian** [comment]
Ok, this issue only happens when you link the library (or use `file:...` in your `package.json`).

When you build the library with `ng build my-lib --prod` and manually copy the dist/my-lib into `node_modules/my-lib` (thus not creating a symlink) the `ng serve` is not throwing an error and everything works.

If you use `npm link`, you can see that the original library folder (`dist/my-lib`) gets changed, also the `package.json` will be modified, for example this is added:

```json
  "scripts": {
    "prepublishOnly": "node --eval \"console.error('ERROR: Trying to publish a package that has been compiled by NGCC. This is not allowed.\\nPlease delete and rebuild the package, without compiling with NGCC, before attempting to publish.\\nNote that NGCC may have been run by importing this package into another project that is being built with Ivy enabled.\\n')\" && exit 1"
  }
```

The problem lies here somewhere. If you publish the package to npm, this won't be an issue.

In summary: `npm link` with custom libraries is currently broken (I think only with 9.x.x?). When `ng serve` is being run, the linked package gets modified and thus broken.

**Comment by 00guille** [comment]
Hi, I have two libraries and a webapp,
One of those libraries (A.K.A. ServiceLib) exports a module with 3 services, perform the update to Angular 9 without problems and consume it in the webapp worked perfectly (locally and from npm).
But the library of graphic components (A.K.A. ComponentsLib) came up with several warnings, but finally I could complete it, but at the time of consuming it locally on the webapp I threw the error:
```
TypeError: Cannot read property 'bindingStartIndex' of null
 at Module.ɵɵelementStart (core.js:20997)
    at SimpleHeaderComponent_Template (cocha-jarvis.js:1526)
    at executeTemplate (core.js:11930)
    at renderView (core.js:11716)
    at renderComponent (core.js:13232)
    at renderChildComponents (core.js:11519)
    at renderView (core.js:11742)
    at renderComponent (core.js:13232)
    at renderChildComponents (core.js:11519)
    at renderView (core.js:11742)
```
Taking into account what @ioulian suggested, I published an alpha version of the ComponentsLib library in npm and consumed it in my webapp and it still goes on the same error.

**Comment by ioulian** [comment]
@00guille I got it working by adding this option to `tsconfig.json` of both the library repository and the main app repository:

```json
  "angularCompilerOptions": {
    "enableIvy": false
  }
```

I haven't tested all the possible configurations (like using Ivy in the main App and not using it in the library or vice versa).

I can now link the dependency locally and it works. Hope this helps for now.

**Comment by 00guille** [comment]
it worked! but the idea is to be able to use ivy with the benefits that it carries

**Comment by mhevery** [comment]
The issue is that somehow when the bundle gets built there are two copies of Ivy in there. Open `vendor.js` and search for `const instructionState`. Notice that it is present twice. Not sure why that is happening.

**Comment by chqy24** [comment]
in your angular app that depends on your local library, set `projects.projectName.architect.build.options.preserveSymlinks` with `true` in `angular.json` will prevent the angular cli from updating linked local library

**Comment by tomeustace** [comment]
@chqy24 thanks, that worked a treat for me.

**Comment by thegnuu** [comment]
I just run into the same issue. Our setup is a little bit more complex, since we are loading the library via fetch from an external source, then require it (electron require in our angular application) and load the library with the following snippet.

``` typescript
const result: any = window.require(pathToFetchedAngularUMDBundle);

this._compiler.compileModuleAsync(result.BusinessLogicModule).then((moduleFactory) => {
  try {
    const moduleRef: any = moduleFactory.create(this._injector);
    const component = moduleRef.instance.getComponent();
    const componentFactory = moduleRef.componentFactoryResolver.resolveComponentFactory(result.BusinessLogicComponent);
    const { instance } = this.container.createComponent(componentFactory, null, moduleRef.injector);
  } catch (err) {
    console.error('error', err);
  }
});
```

If I disable Ivy in both, the library and the application, everything works as expected (took me quite a while to get there to be honest). As soon as Ivy is enabled, I get the error mentioned in this issue. I tried using `preserveSymlinks` which does not change anything.

Has anyone an idea what might help in my setup? Unfortunately it is quite difficult to provide an example, but if it helps I can of course try to create something.

**Comment by thegnuu** [comment]
I created a repo which would allow you to reproduce this issue.

https://github.com/thegnuu/angular-electron

**Comment by thegnuu** [comment]
I was able to resolve my issue... After searching quite a while for a reason I realised that the issue might be caused by different angular links (as you basically said with preserveSymlinks) in my setup. Unfortunately, this option was useless in my case because everything is done after compiling the complete project and externally resolved resources are used.

So I created a small custom `require` which resolves exactly the angular version which is bundled in my app, and now everything works with ivy :)

I know that my use case is really special, and I am aware that this is currently by far not the cleanest solution (regex for requires and stuff like this) but at least it works. Here is my test-loader service for angular (running on electron), maybe this might be useful for someone with a similar setup.

Some of the packages are private, but it should help anyway ;)

``` typescript
import { Injectable, Injector, Compiler, NgModuleRef, ComponentFactory, ViewContainerRef } from '@angular/core';
import { BbitHttpClient, BbitUtils } from '@bbitgmbh/bbit.sdk';
import * as AngularCore from '@angular/core';
import * as AngularCommon from '@angular/common';
import { AppService } from './app.service';

@Injectable({
  providedIn: 'root',
})
export class DynamicLoaderService {
  private _moduleRef: NgModuleRef<any>;
  private _componentFactories: ComponentFactory<any>[];

  constructor(private _compiler: Compiler, private _injector: Injector, private moduleRef: NgModuleRef<any>, private _app: AppService) {}

  public async load(): Promise<any> {
    const env = this._app.state().getEnvironment();
    const filePath = `/${env.organizationKey}/0.0.0.js`;
    const app = BbitUtils.require('electron').remote.app;
    const savePath = app.getPath('appData');
    const path = BbitUtils.require('path');
    const fs = BbitUtils.require('fs');
    const file = path.join(path.resolve(savePath), 'business-logic', filePath);
    const url = 'http://localhost:8080/angular/bundles/business-logic.umd.js';

    // fetch js string from url
    const getRes = await BbitHttpClient.fetch(url);
    if (getRes.status !== 200) {
      const error = `Could not fetch js from ${url} status ${getRes.status}`;
      throw new Error(error);
    }
    let js = await getRes.text();

    // implement custom require for angular loading
    if (!(window as any).blRequire) {
      (window as any).blRequire = (pkg: string): any => {
        switch (pkg) {
          case '@angular/core':
            return AngularCore;
          case '@angular/common':
            return AngularCommon;
          default:
            throw new Error('Business logic loader: module ' + pkg + ' not found');
        }
      };
    }
    js = js.replace(/ require\(/gm, ' window.blRequire(');

    // save file
    const dir = path.dirname(file);
    if (!fs.existsSync(dir)) {
      fs.mkdirSync(dir, { recursive: true });
    }
    fs.writeFileSync(file, js);

    // require dynamic module
    const result = window.require(file);

    return this._compileModule(result);
  }

  private async _compileModule(mod: any): Promise<any> {
    const moduleFactory = await this._compiler.compileModuleAndAllComponentsAsync(mod.BusinessLogicModule);
    this._moduleRef = moduleFactory.ngModuleFactory.create(this._injector);
    this._componentFactories = moduleFactory.componentFactories;
  }

  public createComponent(container: ViewContainerRef, selector: string): any {
    if (!this._componentFactories) {
      throw new Error('Dynamic module is not loaded, please call load(...) first');
    }
    const componentFactory = this._componentFactories.find((o) => o.selector === selector);
    if (!componentFactory) {
      throw new Error('Could not find any component with selector ' + selector);
    }
    const { instance } = container.createComponent(componentFactory, null, this._moduleRef.injector);
    // instance.test = this.testSubject;
  }
}
```

**Comment by kctang** [comment]
This happens when `@NgModule()` is resolved from a different `node_modules` location.

Given this configuration, we can use `my-lib` from `my-app` directly without building `my-lib`:
```
# project directory structure
/
  /my-app
    /node_modules
    /projects
  /my-lib
    /node_modules
    /projects

# my-app's tsconfig.json:
    "paths": {
      "my-lib": [
        "../my-lib/projects/my-lib/src/public-api"
      ]
    }
```

As @mhevery highlighted, the error indicates that somehow there are two copies of Ivy in there. This was due to TypeScript's [module resolution](https://www.typescriptlang.org/docs/handbook/module-resolution.html) approach - it looks for `node_modules` by traversing up the directory of file being compiled.

To fix this, I tell Typescript where to find Angular packages by using:
```
    "paths": {
      "@angular/*": [
        "./node_modules/@angular/*"
      ],
      "my-lib": [
        "../my-lib/projects/my-lib/src/public-api"
      ]
    }
```

Initial test seems to be working. Hope this helps.

**Comment by thegnuu** [comment]
@kctang thank you for your detailed answers, this is more or less what I thought might be the issue. 

Unfortunately due to my dynamic loading from external sources I am not able to use this workaround, but it might be a good solution for someone with a less "complex" setup!

**Comment by bazter** [comment]
Specifying path to angular packages as @kctang mentioned was enough for me, thank you for that. I just wanna point out that the path belongs in the compilerOptions section of app's tsconfig.json file.

```
// app's tsconfig.json
{
  ...
  "compilerOptions": {
    ...
    "paths": {
      "@angular/*": [
        "./node_modules/@angular/*"
      ]
    }
  }
}
```
For now I guess I'll have to create a little schematic for my co-workers to spare them the trouble, but I really hope this gets fixed soon tho.

**Comment by abannsunny** [comment]
I hit the same issue trying to have a multi project setup with an ionic project and an angular project trying to share a common library. 

this answer solved the problem for me but I was wondering if it has any downsides to directly referencing the library like "  "../my-lib/projects/my-lib/src/public-api"" rather than looking for the version in the dist folder.

Because this solves a couple of issue for me which I find very useful, 
1. not able to find all references from the library
2.  "go to definition" you will end up in the dist folder than the actual code. 


> This happens when `@NgModule()` is resolved from a different `node_modules` location.
> 
> Given this configuration, we can use `my-lib` from `my-app` directly without building `my-lib`:
> 
> ```
> # project directory structure
> /
>   /my-app
>     /node_modules
>     /projects
>   /my-lib
>     /node_modules
>     /projects
> 
> # my-app's tsconfig.json:
>     "paths": {
>       "my-lib": [
>         "../my-lib/projects/my-lib/src/public-api"
>       ]
>     }
> ```
> 
> As @mhevery highlighted, the error indicates that somehow there are two copies of Ivy in there. This was due to TypeScript's [module resolution](https://www.typescriptlang.org/docs/handbook/module-resolution.html) approach - it looks for `node_modules` by traversing up the directory of file being compiled.
> 
> To fix this, I tell Typescript where to find Angular packages by using:
> 
> ```
>     "paths": {
>       "@angular/*": [
>         "./node_modules/@angular/*"
>       ],
>       "my-lib": [
>         "../my-lib/projects/my-lib/src/public-api"
>       ]
>     }
> ```
> 
> Initial test seems to be working. Hope this helps.
> 
> Also, I think this approach does not depend on `preserveSymlinks` or `"enableIvy": false`.

**Comment by bazter** [comment]
> I hit the same issue trying to have a multi project setup with an ionic project and an angular project trying to share a common library.
> 
> this answer solved the problem for me but I was wondering if it has any downsides to directly referencing the library like " "../my-lib/projects/my-lib/src/public-api"" rather than looking for the version in the dist folder.
> 
> ...

Not sure about the downsides, but please note the very last paragraph in the [docs](https://angular.io/guide/creating-libraries#building-and-rebuilding-your-library). Using path mapping should be only done in development phase - before building the production build, you should revert it and build the library first - separately, then build the app itself. From my personal experience the dist is much smaller this way.

**Comment by JanMod** [comment]
> I hit the same issue trying to have a multi project setup with an ionic project and an angular project trying to share a common library.
> 
> this answer solved the problem for me but I was wondering if it has any downsides to directly referencing the library like " "../my-lib/projects/my-lib/src/public-api"" rather than looking for the version in the dist folder.
> 
> Because this solves a couple of issue for me which I find very useful,
> 
> 1. not able to find all references from the library
> 2. "go to definition" you will end up in the dist folder than the actual code.
> 
> > This happens when `@NgModule()` is resolved from a different `node_modules` location.
> > Given this configuration, we can use `my-lib` from `my-app` directly without building `my-lib`:
> > ```
> > # project directory structure
> > /
> >   /my-app
> >     /node_modules
> >     /projects
> >   /my-lib
> >     /node_modules
> >     /projects
> > 
> > # my-app's tsconfig.json:
> >     "paths": {
> >       "my-lib": [
> >         "../my-lib/projects/my-lib/src/public-api"
> >       ]
> >     }
> > ```
> > 
> > 
> > As @mhevery highlighted, the error indicates that somehow there are two copies of Ivy in there. This was due to TypeScript's [module resolution](https://www.typescriptlang.org/docs/handbook/module-resolution.html) approach - it looks for `node_modules` by traversing up the directory of file being compiled.
> > To fix this, I tell Typescript where to find Angular packages by using:
> > ```
> >     "paths": {
> >       "@angular/*": [
> >         "./node_modules/@angular/*"
> >       ],
> >       "my-lib": [
> >         "../my-lib/projects/my-lib/src/public-api"
> >       ]
> >     }
> > ```
> > 
> > 
> > Initial test seems to be working. Hope this helps.
> > Also, I think this approach does not depend on `preserveSymlinks` or `"enableIvy": false`.

I also have different workspaces for library and app. 
I mean the fix with TypeScript-paths is kinda straightfoward and I did of course. But somehow I still have the same error message while rendering my component template and importing Modules (Services etc) works fine.

So I just imported in **app.module.ts** my created component from library. like this:
`import { LibComponent } from '../../../../myLib/projects/myLib/src/lib/components/libComponent/lib.component';`
And I still get the same error message. 
I mean how is this possible right here?

**Comment by eivanphils** [comment]
Hey men did you find the solution?

**Comment by M-Abozaid** [comment]
I found out that the problem was that I'm using the `--prod` option in my library build command 
the solution was to 
- remove the `--prod` option
- set `"preserveSymlinks": true` and suggested [here](https://github.com/angular/angular/issues/35586#issuecomment-598067833)
- and enable Ivy ` "enableIvy": true`

**Comment by markkorput** [comment]
Still struggling with this issue, cannot reference local external lib in an Angular app, unless I copy the build of the library from its `dist` folder to the `node_modules` folder of my app. For me the `"preserveSymlinks"` solution doesn't fix it (neither with Ivy enabled or disabled); I keep getting the `Cannot read property 'bindingStartIndex' of null` errors.

Hope this still gets picked up!

**Comment by Xsaza** [comment]
Faced the same issue. I have two angular applications, one is main and second is just exporting ngModule. While first app run-time I load ngModule from second app and try to render component of this ngModule in first app like so:

`const { SharedModule } = await System.import('http://localhost/<path>/main.js');
const moduleAndCompFactory = this.compiler.compileModuleAndAllComponentsSync(SharedModule);
const componentFactory = moduleAndCompFactory.componentFactories[0];
this.selectOptionsHost.viewContainerRef.createComponent(componentFactory);`

Everything is fine untill last line. `preserveSymlinks: true` doesn't work for me and as @kctang said this is 'cause `This happens when @NgModule() is resolved from a different node_modules location`. So is it possible to load component from remote server and render it at all?

**Comment by pillsilly** [comment]
> Specifying path to angular packages as @kctang mentioned was enough for me to fix the error, thank you for that. I just wanna point out that the path belongs in the compilerOptions section of app's tsconfig.json file.
> 
> ```
> // app's tsconfig.json
> {
>   ...
>   "compilerOptions": {
>     ...
>     "paths": {
>       "@angular/*": [
>         "./node_modules/@angular/*"
>       ]
>     }
>   }
> }
> ```
> 
> For now I guess I'll have to create a little schematic for my co-workers to spare them the trouble, but I really hope this gets fixed soon tho.
> 
> Edit: I should probably read [documentation](https://angular.io/guide/creating-libraries#use-typescript-path-mapping-for-peer-dependencies) twice next time.
> 
> Edit2: Oh and specifying path to lib's source code to skip building is actually really useful. Big ups to @kctang for that as well!

this actually fixed my same issue

**Comment by lars-berger** [comment]
Ran into this error when there was a version mismatch of `zone.js` between the library and the app in a pnpm workspace. Library was using `0.12.0` and app was using `0.11.8`.

**Comment by alxhub** [comment]
Closing as the issue is related to having multiple copies of Angular at runtime due to the build structure.

Please feel free to open an issue with a reproduction if you encounter this error for another reason.

**Comment by JoostK** [comment]
I am going to reopen this issue as umbrella issue for multiple copies of `@angular/core`, as I have always been referring to this issue for its duplicates.

**Comment by JeanMeche** [comment]
I've open a feature request on the CLI repo to provide a warning/error when this issue occurs : angular/angular-cli#26774

**Comment by Th3S4mur41** [comment]
I've been facing the same issue in a recent project...
In this case though 
```json
"preserveSymlinks": true,
"paths": {
  "@angular/*": ["./node_modules/@angular/*"]
}
```

Doesn't help?

The difference with the other projects, where it worked is that in this case we are working with angular 17 with the [new build system ](https://angular.io/guide/esbuild#for-new-applications) and [vite](https://vitejs.dev/)

Does this now need a different workaround?

**Comment by huw1676** [comment]
Creating a new library using:
```
ng generate library test-button --standalone=false 
```

Creates a new library with a module, a service, and a component.

The new library's module is then imported into any other Angular project.

The new library's component selector is used within the template of the existing Angular project. 

When this application is then served, the error that can be seen within the console is:
```
TypeError: Cannot read properties of null (reading 'bindingStartIndex')
    at Module.ɵɵelementStart (core.mjs:23018:46)
    at TestButtonComponent_Template (test-button.component.ts:6:5)
    at executeTemplate (core.mjs:12154:9)
    at renderView (core.mjs:15272:13)
    at renderComponent (core.mjs:15219:5)
    at renderChildComponents (core.mjs:15317:9)
    at renderView (core.mjs:15297:13)
    at renderComponent (core.mjs:15219:5)
    at renderChildComponents (core.mjs:15317:9)
    at renderView (core.mjs:15297:13)
```

Angular details:
```
Angular CLI: 17.3.2
Node: 18.18.2
```

This was definitely working in previous versions. 

I now cannot use any new component based libraries. PLEASE FIX!

**Comment by Yabonev** [comment]
For anyone struggling with Angular17 + esbuild.

1. Follow the steps TO THE POINT to introduce esbuild to your app https://v17.angular.io/guide/esbuild#for-existing-applications
2. Remove any node module references in your paths in the application `tsconfig.json`. I Had `  "@angular/*": ["node_modules/@angular/*"],`, which I had to remove. Why? Typescript -> https://www.typescriptlang.org/docs/handbook/modules/reference.html#paths-should-not-point-to-monorepo-packages-or-node_modules-packages

I am tired so this might not be all...

My actions that helped me solve the issue, if the above didn't help.
1. Delete all my code, until I have only a single <div> in my app.component (this helped me understand that's it's a config issue)
2. Create a new angular17 app and check if it's running successfully (it is, truly a config issue)
3. Combine my configs with the new app configs. The new app configs take precedence as it's working
4. See that my app now works with the config from the fresh angular17 barebones app
5. Step by step revert changes until I break it to understand which specific config creates the issue

Overall, 
I'd say I am not happy with this bruteforce method, and I could mainly do it because my app is simple. I could and did get away without understanding how things work. A bad long-term lesson.

Conclusion: Learn.

**Comment by Suprabhatgit** [comment]
> in your angular app that depends on your local library, set `projects.projectName.architect.build.options.preserveSymlinks` with `true` in `angular.json` will prevent the angular cli from updating linked local library

"projects": {
  "your-project-name": {
    "architect": {
      "build": {
        "options": {
          "preserveSymlinks": true,
          // other existing options
        }
      }
    }
  }
}

when you are building an angular library and importing it outside your angular workspace, you have to use this  "preserveSymlinks": true , and it will solve the error 

ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'bindingStartIndex' of null
TypeError: Cannot read property 'bindingStartIndex' of null
![Screenshot (115)](https://github.com/user-attachments/assets/e32e0e32-97eb-4eec-afe0-e9a1ebcd6db9)

**Comment by zahmo** [comment]
We experienced this issue when updating angular to v18, but only got the 'bindingStartIndex' issue when running 
`ng serve`. 
If the dist folder is copied and served from our own http server, this issue does not occur and the page loads correctly.
Another workaround we found that does seem to work, is running it as 
`ng serve --prebundle=false`

**Comment by stad-nico** [comment]
Still got the same error when using an angular library with angular 18.

**Comment by stad-nico** [comment]
Still got the same error when using an angular library v18.
As @Th3S4mur41 said, 

```
"preserveSymlinks": true,
"paths": {
  "@angular/*": ["./node_modules/@angular/*"]
}
```

does not help and neither does @zahmo `ng serve --prebundle=false` work!

**Comment by filiptrivan** [comment]
The two most common tips for this problem are:
 1. Changing the paths array inside tsconfig.json and 
 2. Setting "preserveSymlinks": true inside angular.json. 

For me in Angular 17, I had to `delete` all paths from tsconfig.json but `leave` "preserveSymlinks": true

@Yabonev comment led me to the solution.
